lib/sirius/faculty_semester_weeks_generator.rb

```diff
@@ -0,0 +1,17 @@
+module Sirius
+  module FacultySemesterWeeksGenerator
+
+    module_function
+
+    def generate_semester_weeks_dates(semester)
+      faculty_semester_weeks = SiriusApi::SemesterSchedule.resolve_weeks(semester.starts_at, semester.teaching_ends_at, semester.faculty)
+
+      faculty_semester_weeks = faculty_semester_weeks.select { |week| week.teaching_week != nil }
+
+      weeks_starts = faculty_semester_weeks.map { |week| week.start_date}
+      weeks_ends = faculty_semester_weeks.map { |week| week.end_date}
+
+      [weeks_starts, weeks_ends]
```

Comment:
I think it is better to use an array of tuples `(start_date, end_date)`, because it is already computed and you do not need to compute `end_date` again you need it...